src/azure-cli/azure/cli/command_modules/sql/custom.py

```diff
@@ -450,7 +450,7 @@ def _get_identity_object_from_type(
                     if umiDict is None:
                         umiDict = {identity: UserIdentity()}
                     else:
-                        umiDict[identity] = UserIdentity()
+                        umiDict[identity] = UserIdentity()  # pylint: disable=unsupported-assignment-operation
```

Comment:
This error is triggered because `umiDict` is initialized as `None`.

Instead of disabling this error, a better solution is to initialize `umiDict` as `{}` and always use `umiDict[identity] = UserIdentity()` assignment.